Python/flowgraph.c

```diff
@@ -1515,6 +1515,14 @@ optimize_basic_block(PyObject *const_cache, basicblock *bb, PyObject *consts)
                     */
                 }
                 break;
+            case STORE_FAST:
+                if (opcode == nextop &&
+                    oparg == bb->b_instr[i+1].i_oparg &&
+                    bb->b_instr[i].i_loc.lineno == bb->b_instr[i+1].i_loc.lineno) {
```

Comment:
we should have a test verifying the optimization doesn't occur when linenos are different

Reply:
I will add it!